corporate/lib/stripe.py

```diff
@@ -2920,10 +2923,10 @@ def migrate_customer_to_legacy_plan(
             "tier": CustomerPlan.TIER_SELF_HOSTED_LEGACY,
             # End when the new plan starts.
             "end_date": end_date,
+            "next_invoice_date": end_date,
             # The primary mechanism for preventing charges under this
-            # plan is setting a null `next_invoice_date`, but setting
-            # a 0 price is useful defense in depth here.
-            "next_invoice_date": None,
+            # plan is setting 'invoiced_through' to last ledger_entry below,
+            # but setting a 0 price is useful defense in depth here.
```

Comment:
Running a migration for exisitng customers for this isn't worth it if we can, instead of setting `next_invoice_date` here, we can make `make_end_of_cycle_update` run for plan's `end_date`?

Reply:
If we invoice plans when either `next_invoice_date <= event_time`  or `end_date <= event_time` is true (This is what you're suggesting, right?). Then isn't it the same thing? If we keep `next_invoice_date` = None here but we're invoicing this plan eventually, then I think it's better to set next_invoice_date & then the invoice_plan() function will set the `next_invoice_date` to None finally.

Reply:
I think I agree with @amanagr that it's a more correct rule to say "We process invoicing for a plan when we reach it's next invoice date or its end date". `end_date` being finite with `next_invoice_date` being `None` seems like a valid state for specifying that a plan should never be invoiced again but should end at the appointed time.